content/postgresql/concepts/window-functions/window-functions.md

```diff
@@ -0,0 +1,101 @@
+| Ntile()| `SELECT column_name1, column_name2, column_nameN, ntile (Argument) OVER (PARTITION BY column_name ORDER BY column_name DESC) FROM table_name;`|Returns an integer ranging from 1 to the argument value.|
+|Cume dist()| `SELECT column_name1, column_name2, column_nameN, cume_dist () OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the cumulative distribution.|  
+| Percent rank()| `SELECT column_name1, column_name2, column_nameN, percent_rank () OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the relative rank of the current row, that is the total partition rows - 1.|
+| First Value()|`SELECT column_name1, column_name2, column_nameN, first_value (value) OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the -value- of the first row in a window frame.|
+|Last value()|`SELECT column_name1, column_name2, column_nameN, last_value (value) OVER (PARTITION BY column_name ORDER BY column_name ASC) FROM table_name;`|Returns the -value- of the last row in a window frame.|
+
+## Example
+
+To understand the `Rank()` and `Row_number()` functions, let's consider the following _staff_ table as an example to demonstrate how these window functions work:
+
+| id | employee | adrress | contact | salary | 
+|--- | -----    | -----   | -----   | ----   |
+| 01 | John     | Tokyo   | +81     | 5500   |
+| 02 | Kate     | Beijing | +186    | 5000   |
+| 03 | Tad      | Mumbai  | +91     | 2500   |
+| 04 | Martha   | Tokyo   | +727    | 3500   |
+| 05 | Will     | paris   | +33     | 4000   |
+| 06 | Ruth     | Dubai   | +971    | 2500   |
+| 07 | Kaka     | Mumbai  | +213    | 4500   |
+| 08 | Tolf     | Tokyo   | +44     | 2000   |
+
+From the above _staff_ table, to rank the employees using the `Rank()` function, we select the id, salary, and address columns from the _staff_ table, and then use the `RANK()` window function to assign a rank to each row based on the _salary_ column within each partition defined by the _address_ column.
+
+#Syntax:
+
+`SELECT id, salary, address, Rank() OVER (PARTITION BY address ORDER BY salary DESC) FROM staff;`
+
+Breaking down the syntax:
+- The clause `SELECT`, selects the id, salary, and address colums from the staff table.
+- The `Rank() OVER` part computes the row number of each row.
+- The `(PARTITION BY address ORDER BY salary DESC)` clause actively defines how the window function partitions the data.
+- The statement ends with `FROM staff`, signifying that it operates on the staff table.
+
+#Output:
+
+| id | salary | adrress | rank | 
+|--- | -----  | -----   | ---- |
+| 01 |  5500  | Tokyo   |  1   |
+| 04 |  3500  | Tokyo   |  2   |
+| 08 |  2000  | Tokyo   |  3   |
+| 07 |  4500  | Mumbai  |  1   |
+| 03 |  2500  | Mumbai  |  2   |
+| 02 |  5000  | Beijing |  1   |
+| 05 |  4000  | Paris   |  1   |
+| 06 |  2500  | Dubai   |  1   |
```

Comment:
```suggestion
## Output

| `id` | `salary`| `address` | `rank` | 
|--- | -----  | -----   | ---- |
| 01 |  5500  | Tokyo   |  1   |
| 04 |  3500  | Tokyo   |  2 |
| 08 |  2000  | Tokyo   |  3 |
| 07 |  4500  | Mumbai | 1 |
| 03 |  2500  | Mumbai | 2 |
| 02 |  5000  | Beijing | 1 |
| 05 |  4000  | Paris    | 1 |
| 06 |  2500  | Dubai | 1 |
```
Add a line like this after the subheading `## Output`: 
`This is the output of the above query:`